src/test/java/com/google/firebase/remoteconfig/FirebaseRemoteConfigClientImplTest.java

```diff
@@ -277,12 +305,399 @@ public void testGetTemplateErrorWithRcError() {
         fail("No error thrown for HTTP error");
       } catch (FirebaseRemoteConfigException error) {
         checkExceptionFromHttpResponse(error, ErrorCode.INVALID_ARGUMENT,
-                RemoteConfigErrorCode.INVALID_ARGUMENT, "[INVALID_ARGUMENT]: test error");
+                RemoteConfigErrorCode.INVALID_ARGUMENT, "[INVALID_ARGUMENT]: test error",
+                HttpMethods.GET);
       }
       checkGetRequestHeader(interceptor.getLastRequest());
     }
   }
 
+  // Test getTemplateAtVersion
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testGetTemplateAtVersionWithNullString() throws Exception {
+    client.getTemplateAtVersion(null);
+  }
+
+  @Test
+  public void testGetTemplateAtVersionWithInvalidString() throws Exception {
+    List<String> invalidVersionStrings = ImmutableList
+            .of("", " ", "abc", "t123", "123t", "t123t", "12t3", "#$*&^", "-123", "+123", "123.4");
+
+    for (String version : invalidVersionStrings) {
+      try {
+        client.getTemplateAtVersion(version);
+        fail("No error thrown for invalid version number");
+      } catch (IllegalArgumentException expected) {
+        String message = "Version number must be a non-empty string in int64 format.";
+        assertEquals(message, expected.getMessage());
+      }
+    }
+  }
+
+  @Test
+  public void testGetTemplateAtVersionWithValidString() throws Exception {
+    response.addHeader("etag", TEST_ETAG);
+    response.setContent(MOCK_TEMPLATE_RESPONSE);
+
+    Template receivedTemplate = client.getTemplateAtVersion("24");
+
+    assertEquals(TEST_ETAG, receivedTemplate.getETag());
+    assertEquals(EXPECTED_TEMPLATE, receivedTemplate);
+    assertEquals(1605423446000L, receivedTemplate.getVersion().getUpdateTime());
+    checkGetRequestHeader(interceptor.getLastRequest(), "?versionNumber=24");
+  }
+
+  @Test
+  public void testGetTemplateAtVersionWithEmptyTemplateResponse() throws Exception {
+    response.addHeader("etag", TEST_ETAG);
+    response.setContent("{}");
+
+    Template template = client.getTemplateAtVersion("24");
+
+    assertEquals(TEST_ETAG, template.getETag());
+    assertEquals(0, template.getParameters().size());
+    assertEquals(0, template.getConditions().size());
+    assertEquals(0, template.getParameterGroups().size());
+    assertNull(template.getVersion());
+    checkGetRequestHeader(interceptor.getLastRequest(), "?versionNumber=24");
+  }
+
+  @Test(expected = IllegalStateException.class)
+  public void testGetTemplateAtVersionWithInvalidEtags() throws FirebaseRemoteConfigException {
```

Comment:
Which line is supposed to throw in this test case?

Reply:
Good catch! It should throw on `client.getTemplateAtVersion("24");`. Converted into to two separate tests.